src/index.js

```diff
@@ -270,39 +273,51 @@ class KadDHT {
             return cb(new Error('Failed to lookup key'))
           }
 
-          // we have peers, lets do the actualy query to them
-          const query = new Query(this, key, (peer, cb) => {
-            this._getValueOrPeers(peer, key, (err, rec, peers) => {
-              if (err) {
-                // If we have an invalid record we just want to continue and fetch a new one.
-                if (!(err instanceof errors.InvalidRecordError)) {
-                  return cb(err)
+          // we have peers, lets do the actual query to them
+          const query = new Query(this, key, () => {
+            const pathVals = []
+            paths.push(pathVals)
```

Comment:
Yes, it's initially empty. Contents are pushed into pathVals inside the following function. Should I clarify this?

Reply:
Ok, I understand what you are doing. So, you add the reference and then push to it. Why not pushing it to the `paths` by the end? I think the logic would be more understandable.

Reply:
Sure, can do.

Reply:
Actually, I'm not sure where you're proposing to put it. It can't be after the return statement, or it won't get executed. Do you want the returned function assigned to a variable, then the push, and finally a separate return statement?

Reply:
I think you can do it before:

```js
cb(null, res)
```

that is, inside the return function. What do you think?

Reply:
That would run the push once per query instead of once per path, so we would have lots of duplicates. Unless I'm misunderstanding?

Reply:
oh yes, you are right! It is ok then